src/PolykeyAgent.ts

```diff
@@ -103,7 +107,6 @@ class PolykeyAgent {
     vaultManager,
     notificationsManager,
     sessionManager,
-    grpcServerClient,
```

Comment:
Don't just remove this and then establish it in the agent. Preserve the pattern that we are using. Dependency inject the `rpcServerClient`. Or whatever is relevant here.

Plus you didn't update the types to remove `grpcServerClient`, that's still there.

The original thing here was meant to be the RPC server for the client service. That's why there's also the `grpcServerAgent`.

I'm guessing now when we migration the agent service to to this, would we also have a `rpcServerAgent`?

Furthermore you would have 2 RPC servers, one for the client service and one for the agent service.

And I also think the `webSocketServer` should also be DIed like the rest of it.

Reply:
Note that:

```
    grpcServerClient?: GRPCServer;
```

Still exists in the types here.

Reply:
I missed the `grpcServerClient`, I'll remove it.

As for DI, I can make them both DI-ed.

Reply:
I don't think the rpcServer can be DI-ed. It has a mutual reference to the `PKAgent`.  This means that it needs to be created within the `PolykeyAgent`'s constructor. '

It's cheesy but we can pass the `PKAgent` into it as a promise to be resolved later. that might be the better option here. I think the only problem is, if you don't resolve the promise, the agent stop handler will hang until timeout.

Reply:
This has been fixed, `rpcServerClient` and `webSocketServerClient` can now be dependency injected.